src/Lexer.php

```diff
@@ -346,11 +354,7 @@ private function lexExpression(): void
         }
         // numbers
         elseif (preg_match(self::REGEX_NUMBER, $this->code, $match, 0, $this->cursor)) {
-            $number = (float) $match[0];  // floats
-            if (ctype_digit($match[0]) && $number <= \PHP_INT_MAX) {
-                $number = (int) $match[0]; // integers lower than the maximum
-            }
-            $this->pushToken(Token::NUMBER_TYPE, $number);
+            $this->pushToken(Token::NUMBER_TYPE, 0 + str_replace('_', '', $match[0]));
```

Comment:
@smnandre Let's ignore the phpstan error (`phpstan-baseline.neon`).

Reply:
Done (you may want to check the comment)